### PR TITLE
Added custom post type and utilities

### PR DESCRIPTION
**Pull request by jmbarne3**
**Description**
Adds the document custom post type. Also adds the utility functions we'll need to make sure ACF is installed and activated before trying to register fields, which will be in a future pull request.

**Motivation and Context**
The main purpose of this plugin is to provide this custom post type.

**How Has This Been Tested?**
Changes will be in dev as soon as the job is created and pushed.

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [ ] My change requires an update to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: ucf-document.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Plugin Name: UCF Document Plugin
+Description: Provides a custom post type and taxonomies for managing documents
+Version: 0.1.0
+Author: UCF Web Communications
+License: GPL3
+GitHub Plugin URI: UCF/UCF-Document-Plugin
+*/
+
+namespace UCFDocument;
+use UCFDocument\Document;
```
Comment (cjg89) [inline ucf-document.php]: `use` needs to be updated to reflect the namespace change in your last commit

**Review by cjg89 — CHANGES_REQUESTED** [review]

File: ucf-document.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Function that runs when the plugin
+ * is activated.
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @return void
+ */
+function activate() {
+	Document\register();
```
Comment (cjg89) [inline ucf-document.php]: This is redundant now that the Document registration function is called in the constructor of that class

**Review by cjg89 — CHANGES_REQUESTED** [review]

File: ucf-document.php (hunk 1 above)
Comment (jmbarne3) [reply ucf-document.php]: Updated.

File: ucf-document.php (hunk 2 above)
Comment (jmbarne3) [reply ucf-document.php]: `register` has to be called before the `flush_rewrite_rules` function is called, so a manual call has to be done. Updated to reflect the register function is in a class now.

**Review by cadie — APPROVED** [review]

**Review by cjg89 — APPROVED** [review]

**Review by rjucf — APPROVED** [review]

**Pull request merged**